utils/ftp.py

```diff
@@ -301,9 +303,11 @@ def update_time_difference(self):
             if name == TIMECHECK_FILE_NAME:
                 self.dt = now - time
                 self.dele(TIMECHECK_FILE_NAME)
+                self.cwd("/")
```

Comment:
If the upload dir is non-/, it should be there anyway, no? So basically, you don't need thes cwd here as you entered the directory on the line 297.